docs/cli/ilab-processes.md

```diff
@@ -0,0 +1,64 @@
+# Processes in InstructLab
+
+The ability to detach from processes is crucial to the user experience of InstructLab. However, the concept of multi-processing, process management, and the monitoring of processes is very complex.
+
+It is important to try and add this concept in as simply as possible, expanding on the state reporting, logging, and other features as we go along.
+
+## Phased approach to InstructLab Processes
+
+This document is going to describe phase 1 of implementing processes in InstructLab. Phase 1 is to be described as the "ilab simple process management system". This will depend purely on python packages, PID tracking, and log files to create the experience of detachable processes. The key here is the concept of the UUID, allowing a future REST API to keep track of InstructLab processes using these unique identifiers.
+
+We can re-visit all this in phase 2, when we discuss if we want to utilize something like systemd or a more in-depth process-monitor repo to track processes.
+
+### Phase 1
+
+Phase one would focus on adding the concept of detaching from processes, re-attaching to them, and managing the various artifacts from the processes.
+
+Process management would only apply to `ilab data generate` and `ilab model train` in a first iteration. This would be followed by commands like `ilab model evaluate`, `ilab model serve`, and `ilab model download`. All of these commands have long running processes that would benefit from detachment.
+
+The workflow would allow for:
+
+`ilab data generate -dt` (run a detached generation process)
+`ilab model train -dt` (run a detached training process)
+
+`ilab process list`
+
+```console=
++------------+-------+--------------------------------------+------------------------------------------------------------------------------------------------------------------+----------+
+| Type       | PID   | UUID                                 | Log File                                                                                                         | Runtime  |
++------------+-------+--------------------------------------+------------------------------------------------------------------------------------------------------------------+----------+
+| Generation | 39832 | 82d00a5b-5ed5-4cfd-9a75-a87e4f420b27 | /Users/charliedoern/.local/share/instructlab/logs/generation/generation-82d00a5b-5ed5-4cfd-9a75-a87e4f420b27.log | 69:26:28 |
```

Comment:
Having `Type` like this would require some kind of mapping from the command to these descriptions - maybe it would be simpler to simply map to the command, i.e. instead of `Generation` do `data generate`?

Reply:
I already have this implemented, either way we would have to pass an argument to the process creation code. I think passing just `Generation` is cleaner.